### ISSUE TITLE
GEOS abuses MAPL_TICE: Worth fixing?

### ISSUE THREAD

**Issue by mathomp4** [GEOS-ESM/GEOSgcm#271]
In talking with @sanAkel today, we noticed that we "abuse" `MAPL_TICE` in GEOS. Namely, in `MAPL_Constants.F90`:
```fortran
real, parameter, public :: MAPL_TICE   = 273.16                 ! K
```
This is the triple point of water, and is correct. But in GEOS we also use it to convert to/from Kelvin, like in Catch GC:
```fortran
6039:  tc(:,FSNW) = tc(:,FSNW) + MAPL_TICE ! Convert to K
```
This is not correct. The triple point of water is 273.16 K or 0.01 ºC. But 0 ºC is 273.15 K. So, we should probably have a:
```fortran
real, parameter, public :: MAPL_KELVIN   = 273.15                 ! K
```
that is used when we want to convert Celsius to Kelvin and vice versa.

The problem is that we'd need to do a serious look at all uses of `MAPL_TICE` in GEOS. This would be non-zero-diff and we'd probably need the approvals of @wmputman for the GCM and @gmao-rreichle for the LDAS... (not sure if the ADAS directly cares, or if all the TICE-Kelvin conversion is from the GCM). Note this would also mean fixing up `KELVIN` in the FMS constants as well (which reflects the GEOS `MAPL_TICE`).

The real question for the scientists is: Should this be done? It's an *instant* non-zero-diff for sure, but it seems more "correct" to me?

**Comment by sanAkel** [comment]
From what I can say for:
>(not sure if the ADAS directly cares, or if all the TICE-Kelvin conversion is from the GCM).

it does not have a "global" constant, every instance is declared and used as a "new variable" see for e.g.,
https://github.com/GEOS-ESM/GEOSana_GridComp/blob/cb564bf830b4beb4ff80e9c5363f1ee596b690ce/GEOSaana_GridComp/GSI_GridComp/clw_mod.f90#L1291

https://github.com/GEOS-ESM/GEOSana_GridComp/blob/cb564bf830b4beb4ff80e9c5363f1ee596b690ce/GEOSaana_GridComp/GSI_GridComp/gsdcloudanalysis.F90#L948

@rtodling may want to say something.

**Comment by gmao-rreichle** [comment]
Yes, there's a distinction between 0 deg C and the triple point of water.  But the difference is negligible compared to the uncertainties in the land model's representation of physical processes.  Soil water doesn't freeze at exactly 0 deg C.  Rather, it's somewhere around -2 deg C, if memory serves me right, and the freezing point of soil water probably varies by much more than the 0.01 deg C difference between 0 deg C and MAPL_TICE.  So from the hydrologist's perspective, the latter difference is a bit academic.  But in my previous life, I studied physics and have no objections to correcting the issue.
The land model and GEOSldas should be reasonably well coded and exclusively use MAPL_TICE, so it'd probably be a simple change.
cc: @rdkoster
